qa/sc_evm_pause_forging.py

```diff
@@ -0,0 +1,170 @@
+            arg2 = ['']
+
+        return start_sc_nodes(
+            self.number_of_sidechain_nodes,
+            self.options.tmpdir, extra_args=[['-max_hist_rew_len', str(101)] + arg1, arg2],
+            binary=[EVM_APP_BINARY] * 2, auth_api_key=self.API_KEY)
+
+    def run_test(self):
+
+        sc_node_1 = self.sc_nodes[0]
+        sc_node_2 = self.sc_nodes[1]
+        mc_node = self.nodes[0]
+
+        # Do FT of some Zen to SC Node 2
+        evm_address_sc_node_2 = sc_node_2.wallet_createPrivateKeySecp256k1()["result"]["proposition"]["address"]
+
+        ft_amount_in_zen = 100
+
+        forward_transfer_to_sidechain(self.sc_nodes_bootstrap_info.sidechain_id,
+                                      mc_node,
+                                      evm_address_sc_node_2,
+                                      ft_amount_in_zen,
+                                      mc_return_address=mc_node.getnewaddress(),
+                                      generate_block=False)
+        self.sync_all()
+
+        # Generate MC block and SC block
+        mc_node.generate(1)
+        generate_next_block(sc_node_1, "first node", force_switch_to_next_epoch=True)
+        self.sc_sync_all()
+
+        # Create forger stake with some Zen for SC node 2
+        sc2_blockSignPubKey = sc_node_2.wallet_createPrivateKey25519()["result"]["proposition"]["publicKey"]
+        sc2_vrfPubKey = sc_node_2.wallet_createVrfSecret()["result"]["proposition"]["publicKey"]
+
+        forger_stake_amount = ft_amount_in_zen - 0.01 # for gas usage
+
+        makeForgerStakeJsonRes = ac_makeForgerStake(sc_node_2, evm_address_sc_node_2, sc2_blockSignPubKey,
+                                                    sc2_vrfPubKey,
+                                                    convertZenToZennies(forger_stake_amount))
+        if "result" not in makeForgerStakeJsonRes:
+            fail("make forger stake failed: " + json.dumps(makeForgerStakeJsonRes))
+        else:
+            logging.info("Forger stake created: " + json.dumps(makeForgerStakeJsonRes))
+        self.sc_sync_all()
+
+        # Generate SC block
+        generate_next_block(sc_node_1, "first node")
+
+        # we now have 2 stakes, one from creation and one just added
+        stakeList = sc_node_1.transaction_allForgingStakes()["result"]['stakes']
+        assert_equal(2, len(stakeList))
+
+        # Generate SC blocks on SC node 1 for switching 2 consensus epochs
+        generate_next_block(sc_node_1, "first node", force_switch_to_next_epoch=True)
+        generate_next_block(sc_node_1, "first node", force_switch_to_next_epoch=True)
+        # mc_node.generate(1)
+        self.sc_sync_all()
+
+        # verify that before the fork point we can forge more than 100 sc blocks without including a mc block ref
+        NUM_OF_BLOCKS = 102
+
+        for i in range(NUM_OF_BLOCKS):
+            generate_next_block(sc_node_1, "second node")
+            self.sc_sync_all()
+
+        # reach the SHANGHAI fork
```

Comment:
This comment should be updated because Shanghai fork is 1.2

Reply:
In order to rule out ambiguity I removed the comment

Reply:
Actually, Shanghai fork is Fork 1.3, I was wrong. I agree on removing the comment